### PR TITLE
Benji/minoruifixes

### PR DESCRIPTION
**Pull request by Benjamindodgson**

### CODE REVIEW

File: Shared/Flows/Toast/ToastBannerView.swift [hunk 1]
```diff
@@ -11,10 +11,12 @@ import Localization
 
 class ToastBannerView: ToastView {
     
-    private let minimumHeight: CGFloat = Theme.ContentOffset.standard.value.doubled + 44
+    private static let personHeight: CGFloat = 38
```
Comment (Benjamindodgson) [inline Shared/Flows/Toast/ToastBannerView.swift:14]: Improved toast layout

File: iOS/Flows/Conversation/Attachments/AttachmentsCollectionViewDataSource.swift [hunk 2]
```diff
@@ -90,14 +81,19 @@ class AttachmentsCollectionViewDataSource: CollectionViewDataSource<AttachmentsC
                                                                     for: indexPath,
                                                                     item: option)
             default:
-                return collectionView.dequeueConfiguredReusableCell(using: self.optionConfig,
-                                                                    for: indexPath,
-                                                                    item: option)
+                let lastIndex = self.snapshot().numberOfItems(inSection: section) - 1
```
Comment (Benjamindodgson) [inline iOS/Flows/Conversation/Attachments/AttachmentsCollectionViewDataSource.swift:84]: Consistent layout using the new layout cell

Comment (Benjamindodgson) [reply iOS/Flows/Conversation/Attachments/AttachmentsCollectionViewDataSource.swift:84]: OptionCell

File: iOS/Flows/Conversation/Attachments/AttachmentsCollectionViewLayout.swift [hunk 3]
```diff
@@ -55,39 +55,44 @@ class AttachmentCollectionViewLayout: UICollectionViewCompositionalLayout {
...
+                                                                trailing: sectionInset)
                 
-                let headerItemSize = NSCollectionLayoutSize(widthDimension: .fractionalWidth(1), heightDimension: .estimated(Theme.ContentOffset.xtraLong.value))
-                let headerItem = NSCollectionLayoutBoundarySupplementaryItem(layoutSize: headerItemSize, elementKind: UICollectionView.elementKindSectionHeader, alignment: .top)
-                headerItem.contentInsets = NSDirectionalEdgeInsets(top: 0,
-                                                                   leading: 0,
-                                                                   bottom: 0,
-                                                                   trailing: 0)
-                section.boundarySupplementaryItems = [headerItem]
+                let backgroundItem = NSCollectionLayoutDecorationItem.background(elementKind: SectionBackgroundView.kind)
```
Comment (Benjamindodgson) [inline iOS/Flows/Conversation/Attachments/AttachmentsCollectionViewLayout.swift:82]: I wish there was an easy to package this but not really...

File: iOS/Flows/Threads/ThreadViewController.swift [hunk 4]
```diff
@@ -117,7 +117,7 @@ class ThreadViewController: DiffableCollectionViewController<MessageSequenceSect
         self.view.insertSubview(self.blurView, belowSubview: self.collectionView)
         self.view.addSubview(self.parentMessageView)
         
-        self.addChild(viewController: self.detailVC)
+        self.addChild(viewController: self.detailVC, toView: self.parentMessageView)
```
Comment (Benjamindodgson) [inline iOS/Flows/Threads/ThreadViewController.swift:120]: Improved thread layout

File: iOS/Flows/Threads/ThreadViewController.swift [hunk 5]
```diff
@@ -198,13 +197,32 @@ class ThreadViewController: DiffableCollectionViewController<MessageSequenceSect
...
         let threadLayout = self.threadCollectionView.threadLayout
         threadLayout.itemHeight
-        = MessageContentView.bubbleHeight + old_MessageDetailView.height + Theme.ContentOffset.short.value
-
+        = MessageContentView.bubbleHeight + old_MessageDetailView.height + Theme.ContentOffset.long.value
+        
         switch state {
         case .read:
-            threadLayout.spacingKeyPoints = [0, 40, 74, 86]
+            let topOfStack = UIWindow.topWindow()!.safeAreaInsets.top + PullView.height + threadLayout.itemHeight + 20
```
Comment (Benjamindodgson) [inline iOS/Flows/Threads/ThreadViewController.swift:204]: sometimes the insets are incorrect when this is called.

File: iOS/Flows/Threads/ThreadViewController.swift [hunk 6]
```diff
@@ -198,13 +197,32 @@ class ThreadViewController: DiffableCollectionViewController<MessageSequenceSect
...
+            threadLayout.topOfStackY = topOfStack
+            threadLayout.spacingKeyPoints = [0, 20, 40, 64]
+            
+            UIView.animate(withDuration: Theme.animationDurationFast) {
+                self.parentMessageView.alpha = 1.0
+            } completion: { _ in
+                self.view.bringSubviewToFront(self.parentMessageView)
```
Comment (Benjamindodgson) [inline iOS/Flows/Threads/ThreadViewController.swift:211]: Useful for moving views around.

**Review by MartinJibber — APPROVED** [review]

**Pull request merged**